File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
	vector<int> answer;
	int arr[26]{};
	for (string now : keymap) {
		for (int i = 0; i < now.size(); i++) {
			int cnt = i + 1;
			if (!arr[now[i] - 'A'])
				arr[now[i] - 'A'] = cnt;
			else
				arr[now[i] - 'A'] = min(arr[now[i] - 'A'], cnt);
		}
	}
	for (string now : targets) {
		int s = 0;
		for (char i : now) {
			if (!arr[i - 'A']) {
				s = -1;
				break;
			}
			s += arr[i - 'A'];
		}
		answer.push_back(s);
	}
	return answer;
}

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []

    arr = [ 0 for i in range(26) ]
    for i in keymap:
        for j in range(len(i)):
            n = j+1
            if arr[ord(i[j])-ord('A')] == 0:
                arr[ord(i[j])-ord('A')] = n
            else:
                arr[ord(i[j])-ord('A')] = min(arr[ord(i[j])-ord('A')], n)

    for i in targets:
        s = 0
        for j in i:
            if arr[ord(j)-ord('A')] == 0:
                s = -1
                break
            s += arr[ord(j)-ord('A')]
        answer.append(s)

    return answer
                